patternfly/patternfly-elements
# chore(ci): Updating CI to allow tests on docs updates and clean up label logic

### What has changed and why

- Add PR label/unlabel to automerge task; include ready to merge as a requirement for automerge to work
- Remove pull request review as a firing task for the automerge
- Validate conventional commit syntax before adding the ready to merge label (cc-format.yml)
- Include blocked & needs: & ready: labels before adding the ready to merge label
- Remove paths-ignore from the build task because it can block PRs from merging if they only update documentation
- Separate label workflows to limit the run for the labeler tool to opened & synchronized

### Testing instructions

- [x] Validate labelling
  1. Open a fork branch with this PR as the base.  Commit only a markdown file update.  The build should be kicked off for that PR.
  2. Flag that fork branch PR as `on hold` then change it to draft state.  `work in progress` label should _not_ be added.
  3. Set the `ready: code review` label on the PR and then approve it.  `ready to merge` label should _not_ be added.
  4. Remove the label and then approve the PR again.  The `ready to merge` label should be added.
  5. Confirm that the labelling/unlabelling is not causing an infinite loop with the label task.

### Ready-for-merge Checklist




### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
